ReactApp/src/sections/popUps/changePasswordPopup.js

```diff
@@ -1,83 +1,99 @@
-        throw LOGIN.ILLEGAL_PASSWORD
-      await auth.changePassword(user.username, changePasswordValues.newPassword);
+      await auth.changePassword(user.username, values.newPassword);
       router.push('/');
     } catch (err) {
-      setErrorMessage(err)
+      setErrorMessage(err.message);
+      setSubmitting(false);
     }
   };
 
   return (
     isOpen ? (
-     <div className="popup">
-      <div onClick={closePopup} style={overlayStyle}></div>
-      <div className="popup-content" style={{ ...popupStyle, width: '30%' }}>
-        <Typography variant="h5" component="h2" gutterBottom padding={2}>
-          {LOGIN.CHANGE_PASSWORD_TITLE}
-        </Typography>
-        <Stack spacing={1}>
-          <TextField
-            margin="dense"
-            id="newPassword"
-            label={LOGIN.NEW_PASSWORD_LABEL}
-            type="password"
-            fullWidth
-            value={changePasswordValues.newPassword}
-            onChange={(e) => setChangePasswordValues({ ...changePasswordValues, newPassword: e.target.value })}
-          />
-          <TextField
-            margin="dense"
-            id="confirmNewPassword"
-            label={LOGIN.CONFIRM_NEW_PASSWORD_LABEL}
-            type="password"
-            fullWidth
-            value={changePasswordValues.confirmNewPassword}
-            onChange={(e) => setChangePasswordValues({ ...changePasswordValues, confirmNewPassword: e.target.value })}
-          />
-          <Stack direction="row" spacing={2} width="100%">
-            <Button onClick={closePopup}>
-              {LOGIN.CANCEL_BUTTON}
-            </Button>
-            <Button type="submit" onClick={handleSubmit}>
-              {LOGIN.CHANGE_PASSWORD_BUTTON}
-            </Button>
-          </Stack>
-          <ErrorMessage message={errorMessage}/>
-        </Stack>
+      <div className="popup">
+        <div onClick={closePopup} style={overlayStyle}></div>
+        <div className="popup-content" style={{ ...popupStyle, width: '30%' }}>
+          <Typography variant="h5" component="h2" gutterBottom padding={2}>
+            {LOGIN.CHANGE_PASSWORD_TITLE}
+          </Typography>
+          <Formik
+            initialValues={{ newPassword: '', confirmNewPassword: '' }}
+            validationSchema={validationSchema}
+            onSubmit={handleSubmit}
+          >
+            {({ isSubmitting, handleChange, handleBlur, values, touched, errors }) => (
+              <Form>
+                <Stack spacing={1}>
+                  <TextField
+                    margin="dense"
+                    id="newPassword"
+                    name="newPassword"
+                    label={LOGIN.NEW_PASSWORD_LABEL}
+                    type="password"
+                    fullWidth
+                    value={values.newPassword}
+                    onChange={handleChange}
+                    onBlur={handleBlur}
+                    error={touched.newPassword && !!errors.newPassword}
```

Comment:
why !! ?

Reply:
in order to have it submited with an enter press